credentials/src/main/java/me/uport/sdk/credentials/Credentials.kt

```diff
@@ -151,9 +154,113 @@ class Credentials(
+     * @param audience [**optional**] the intended audience of this credential.
+     *      This results in the `aud` field of the JWT. If it is `null`, no audience will be set for the JWT.
+     * @param id [**optional**] the ID of this credential.
+     *      This becomes the `jti` field in the resulting JWT.
+     */
+    suspend fun createVerifiableCredential(
+        subject: String,
+        credential: CredentialParams,
+        notValidBefore: Long = clock.nowMs() / 1000L,
+        validityPeriod: Long = -1L,
+        audience: String? = null,
+        id: String? = null
+    ): String {
+
+        val payload = mutableMapOf<String, Any>()
+        payload["sub"] = subject
+
+        //add defaults if they are not set
+        val processedCredential = credential.copy(
+            context = credential.context.toMutableSet().apply { add("https://www.w3.org/2018/credentials/v1") }.toList(),
+            type = credential.type.toMutableSet().apply { add("VerifiableCredential") }.toList()
+        )
+
+        payload["vc"] = processedCredential
+        payload["nbf"] = notValidBefore
+        payload["iat"] = notValidBefore //for backward compatibility
+        if (validityPeriod >= 0) {
+            val exp = notValidBefore + validityPeriod
+            payload["exp"] = exp
+        }
+        if (audience != null) {
+            payload["aud"] = audience
+        }
+        if (id != null) {
+            payload["jti"] = id
+        }
+
+        return this.signJWT(
+            payload = payload,
+            expiresInSeconds = validityPeriod,
+            algorithm = ES256K
+        )
+    }
+
+    /**
+     * Creates a W3C compliant [verifiable presentation](https://w3c.github.io/vc-data-model/#presentations)
+     * serialized as a JWT.
+     *
+     * @param vp encapsulates the core presentation parameters
+     * @param notValidBefore the `issuanceDate` as a unix timestamp (seconds). Defaults to the clock time
+     * (translates to `nbf` and `iat`)
+     * @param validityPeriod the number of seconds of validity of the resulting JWT. (influences `exp`)
+     * @param audience the intended audience of the resulting JWT (translates to `aud`)
+     * @param id an optional ID of the presentation (translates to `jti`)
+     */
+    suspend fun createPresentation(
+        vp: PresentationParams,
+        notValidBefore: Long = clock.nowMs() / 1000L,
+        validityPeriod: Long = -1L,
+        audience: String? = null,
+        id: String? = null
+    ): String {
+        val payload = mutableMapOf<String, Any>()
+
+        val processedPresentation = vp.copy(
+            context = vp.context.toMutableSet().apply { add("https://www.w3.org/2018/credentials/v1") }.toList(),
+            type = vp.type.toMutableSet().apply { add("VerifiablePresentation") }.toList()
+        )
+        payload["vp"] = processedPresentation
+        payload["nbf"] = notValidBefore
+        payload["iat"] = notValidBefore //for backward compatibility
```

Comment:
Can this be removed now that backwards compatibility is not a priority?

Reply:
If so, should we also remove the use of `iat` from `did-jwt` in favor of `nbf`? @awoie

Reply:
```suggestion
        payload["iat"] = null
```

Reply:
I removed the `iat` field from the JWT when building VCs and VPs